### PR TITLE
Add topas mock, resolves #638 

### LINKED ISSUE #638: add Mock executable for TOPAS
Similar to https://github.com/DataMedSci/pymchelper/issues/637 but for TOPAS.
Instead of binary output use real CSV output file

### PR DESCRIPTION
**Pull request by hendzeld**

### CODE REVIEW

File: tests/res/mocks/topas_minimal/minimal.txt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+s:Sc/FluenceBPprotonsXY/IfOutputFileAlreadyExists = "Overwrite"
+s:Sc/FluenceBPprotonsXY/OutputFile = "fluence_bp_protons_xy"
+sv:Sc/FluenceBPprotonsXY/OnlyIncludeIfIncidentParticlesNamed = 1 "proton"
+
+s:Sc/FluenceBPprotonsXY2/Quantity                  = "Fluence"
+s:Sc/FluenceBPprotonsXY2/WeightBy                  = "Track"
+s:Sc/FluenceBPprotonsXY2/Component                 = "PhantomScoringCoarseXY2"
+sv:Sc/FluenceBPprotonsXY2/Report = 2 "Mean" "Standard_Deviation"
+s:Sc/FluenceBPprotonsXY2/IfOutputFileAlreadyExists = "Overwrite"
+s:Sc/FluenceBPprotonsXY2/OutputFile = "fluence_bp_protons_xy2"
```
Comment (grzanka) [inline tests/res/mocks/topas_minimal/minimal.txt:74]:
```suggestion
s:Sc/FluenceBPprotonsXY2/OutputFile = "fluence_bp_protons_xy2.csv"

```

File: tests/res/mocks/topas_minimal/minimal.txt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#u:So/Demo/BeamEnergySpread = 1.018
+#d:So/Demo/OnlyIncludeParticlesWithInitialKEBelow = 58.55 MeV
+
+
+s:Sc/FluenceBPprotonsXY/Quantity                  = "Fluence"
+s:Sc/FluenceBPprotonsXY/WeightBy                  = "Track"
+s:Sc/FluenceBPprotonsXY/Component                 = "PhantomScoringCoarseXY"
+sv:Sc/FluenceBPprotonsXY/Report = 2 "Mean" "Standard_Deviation"
+s:Sc/FluenceBPprotonsXY/IfOutputFileAlreadyExists = "Overwrite"
+s:Sc/FluenceBPprotonsXY/OutputFile = "fluence_bp_protons_xy"
```
Comment (hendzeld) [inline tests/res/mocks/topas_minimal/minimal.txt:66]:
```suggestion
s:Sc/FluenceBPprotonsXY/OutputFile = "fluence_bp_protons_xy.csv"
```

File: tests/res/mocks/topas_minimal/minimal.txt (hunk 1 above)
Comment (hendzeld) [reply tests/res/mocks/topas_minimal/minimal.txt]: I verified it now.
It creates file *fluence_bp_protons_xy2.csv.csv*

obraz

Comment (grzanka) [reply tests/res/mocks/topas_minimal/minimal.txt:74]: Thanks, good to know !

File: tests/res/mocks/topas_minimal/minimal.txt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+d:So/Demo/BeamPositionCutoffX = 2. cm # X extent of position (if Flat or Gaussian)
+d:So/Demo/BeamPositionCutoffY = 2. cm # Y extent of position (if Flat or Gaussian)
+s:So/Demo/BeamAngularDistribution = "None"
+i:So/Demo/NumberOfHistoriesInRun = Ts/ShowHistoryCountAtInterval * 10
+
+# Parisi Beam model
+#d:So/Demo/BeamEnergy = 58.4 MeV
+#u:So/Demo/BeamEnergySpread = 1.018
+#d:So/Demo/OnlyIncludeParticlesWithInitialKEBelow = 58.55 MeV
+
```
Comment (grzanka) [inline tests/res/mocks/topas_minimal/minimal.txt:59]:
```suggestion
```

File: tests/mock/test_topas_minimal.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        env["PATH"] = str(path)
+    return env
+
+
+def from_csv(path: Path) -> np.ndarray:
+    """Reads csv file and returns numpy array"""
+    lines = np.genfromtxt(path, delimiter=',')
+    x = lines[:, 0].astype(int).max() + 1
+    y = lines[:, 1].astype(int).max() + 1
+    z = lines[:, 2].astype(int).max() + 1
```
Comment (grzanka) [inline tests/mock/test_topas_minimal.py:85]: Add a comment explaining what is this code doing. What is x,y,z ? Position where we score fluence or the fluence values ?

File: tests/mock/test_topas_minimal.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    data = np.delete(lines, [0, 1, 2, 4], axis=1)
```
Comment (grzanka) [inline tests/mock/test_topas_minimal.py:86]: Add comment explaining this removal

File: tests/mock/test_topas_minimal.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+@pytest.mark.skipif(os.name == 'nt', reason="Windows not supported")
+@pytest.mark.parametrize('output_file', ["fluence_bp_protons_xy.csv", "fluence_bp_protons_xy2.csv"])
+def test_topas_mock(tmp_path: Path, output_file: str, expected_results: dict, topas_mock: Path):
+    """Test topas mock script"""
+    assert topas_mock.exists(), "Topas script does not exist"
+
+    env = append_path_to_environ(topas_mock.parent)
+    call("topas", cwd=tmp_path, env=env)  # skipcq: BAN-B607
+    output_files = [f.name for f in list(tmp_path.glob("*"))]
```
Comment (grzanka) [inline tests/mock/test_topas_minimal.py:55]:
```suggestion
    output_files = [f.name for f in list(tmp_path.glob("*.csv"))]
```

Shouldn't we go through CSV files only ?

**Review by grzanka — CHANGES_REQUESTED** [review]
see comments

File: tests/mock/test_topas_minimal.py (hunk 5 above)
Comment (hendzeld) [reply tests/mock/test_topas_minimal.py]: It was unnecessarily complicated, please check now.

File: tests/mock/test_topas_minimal.py (hunk 4 above)
Comment (hendzeld) [reply tests/mock/test_topas_minimal.py]: Is it ok now?

**Review by grzanka — APPROVED** [review]

**Pull request merged**